From 22927d4bf2d83e4642f08a7f338ae5dc3a6a60fa Mon Sep 17 00:00:00 2001
From: Philippe Gerum <philippe.gerum@exail.com>
Date: Sun, 29 Oct 2023 11:51:59 +0100
Subject: [PATCH 0829/1013] evl/wait: remove useless dynamic lock key

Using a locally-defined static key for the lock serializing accesses
to a wait queue is enough and non-ambiguous.

Signed-off-by: Philippe Gerum <philippe.gerum@exail.com>
---
 include/evl/wait.h   | 26 +++++---------------------
 kernel/evl/monitor.c |  3 +--
 kernel/evl/wait.c    | 19 ++-----------------
 3 files changed, 8 insertions(+), 40 deletions(-)

diff --git a/include/evl/wait.h b/include/evl/wait.h
index ba58c0210e89..adb3e6d185f4 100644
--- a/include/evl/wait.h
+++ b/include/evl/wait.h
@@ -20,7 +20,6 @@ struct evl_thread;
 
 struct evl_wait_channel {
 	hard_spinlock_t lock;
-	struct lock_class_key lock_key;	/* lockdep disambiguation */
 	struct evl_thread *owner;
 	bool (*requeue_wait)(struct evl_wait_channel *wchan,
 			struct evl_thread *waiter);
@@ -28,27 +27,15 @@ struct evl_wait_channel {
 	const char *name;
 };
 
-#ifdef CONFIG_LOCKDEP
-struct evl_lock_key_addr {
-	struct lock_class_key *addr;
-};
-#define __EVL_LOCK_KEY_ADDR_INITIALIZER  (struct evl_lock_key_addr){ .addr = NULL }
-#else
-struct evl_lock_key_addr { };
-#define __EVL_LOCK_KEY_ADDR_INITIALIZER  (struct evl_lock_key_addr){ }
-#endif
-
 struct evl_wait_queue {
 	int flags;
 	struct evl_clock *clock;
 	struct evl_wait_channel wchan;
-	struct evl_lock_key_addr lock_key_addr;
 };
 
 #define EVL_WAIT_INITIALIZER(__name) {					\
 		.flags = EVL_WAIT_PRIO,					\
 		.clock = &evl_mono_clock,				\
-		.lock_key_addr = __EVL_LOCK_KEY_ADDR_INITIALIZER,	\
 		.wchan = {						\
 			.lock = __HARD_SPIN_LOCK_INITIALIZER((__name).wchan.lock), \
 			.pi_serial = 0,					\
@@ -122,20 +109,17 @@ void __evl_init_wait(struct evl_wait_queue *wq,
 		struct evl_clock *clock,
 		int flags,
 		const char *name,
-		struct lock_class_key *lock_key);
-
-#define evl_init_named_wait(__wq, __clock, __flags, __name)	\
-	__evl_init_wait(__wq, __clock, __flags, __name, NULL)
-
-#define evl_init_wait(__wq, __clock, __flags)	\
-	evl_init_named_wait(__wq, __clock, __flags, #__wq)
+		struct lock_class_key *key);
 
-#define evl_init_wait_on_stack(__wq, __clock, __flags)	\
+#define evl_init_wait(__wq, __clock, __flags)				\
 	do {								\
 		static struct lock_class_key __key;			\
 		__evl_init_wait(__wq, __clock, __flags, #__wq, &__key); \
 	} while (0)
 
+#define evl_init_wait_on_stack(__wq, __clock, __flags)	\
+	evl_init_wait(__wq, __clock, __flags)
+
 void evl_destroy_wait(struct evl_wait_queue *wq);
 
 struct evl_thread *evl_wait_head(struct evl_wait_queue *wq);
diff --git a/kernel/evl/monitor.c b/kernel/evl/monitor.c
index b5ab70d106e3..aac19df39f7a 100644
--- a/kernel/evl/monitor.c
+++ b/kernel/evl/monitor.c
@@ -1058,8 +1058,7 @@ monitor_factory_build(struct evl_factory *fac, const char __user *u_name,
 		INIT_LIST_HEAD(&mon->events);
 		break;
 	case EVL_MONITOR_EVENT:
-		evl_init_named_wait(&mon->wait_queue, clock, EVL_WAIT_PRIO,
-				evl_element_name(&mon->element));
+		evl_init_wait(&mon->wait_queue, clock, EVL_WAIT_PRIO);
 		state->u.event.gate_offset = EVL_MONITOR_NOGATE;
 		atomic_set(&state->u.event.value, attrs.initval);
 		evl_init_poll_head(&mon->poll_head);
diff --git a/kernel/evl/wait.c b/kernel/evl/wait.c
index 5305a0a1b132..26d48cee3b55 100644
--- a/kernel/evl/wait.c
+++ b/kernel/evl/wait.c
@@ -25,18 +25,8 @@ void __evl_init_wait(struct evl_wait_queue *wq,
 	wq->wchan.name = name;
 	INIT_LIST_HEAD(&wq->wchan.wait_list);
 	raw_spin_lock_init(&wq->wchan.lock);
-#ifdef CONFIG_LOCKDEP
-	if (lock_key) {
-		wq->lock_key_addr.addr = lock_key;
-		lockdep_set_class_and_name(&wq->wchan.lock, lock_key, name);
-	} else {
-		lock_key = &wq->wchan.lock_key;
-		wq->lock_key_addr.addr = lock_key;
-		lockdep_register_key(lock_key);
-		lockdep_set_class_and_name(&wq->wchan.lock, lock_key, name);
-		might_hard_lock(&wq->wchan.lock);
-	}
-#endif
+	lockdep_set_class_and_name(&wq->wchan.lock, lock_key, name);
+	might_hard_lock(&wq->wchan.lock);
 }
 EXPORT_SYMBOL_GPL(__evl_init_wait);
 
@@ -44,11 +34,6 @@ void evl_destroy_wait(struct evl_wait_queue *wq)
 {
 	evl_flush_wait(wq, EVL_T_RMID);
 	evl_schedule();
-#ifdef CONFIG_LOCKDEP
-	/* Drop dynamic key. */
-	if (wq->lock_key_addr.addr == &wq->wchan.lock_key)
-		lockdep_unregister_key(&wq->wchan.lock_key);
-#endif
 }
 EXPORT_SYMBOL_GPL(evl_destroy_wait);
 
-- 
2.30.2

